File: turtle-web/src/app/services/tax-info.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { TaxInfo } from '../models/tax-info.model';
import { environment } from '../../environments/environment';
import { Page } from '../models/page.model';
import { ApiResponse } from '../models/api.model';

@Injectable({
  providedIn: 'root'
})
export class TaxInfoService {
  private apiUrl = `${environment.apiUrl}/tax-infos`;

  constructor(private http: HttpClient) {}

  getTaxInfos(params: { page: number; size: number; search?: string; active?: boolean }): Observable<ApiResponse<Page<TaxInfo>>> {
    let httpParams = new HttpParams()
      .set('page', params.page.toString())
      .set('size', params.size.toString());

    if (params.search) {
      httpParams = httpParams.set('search', params.search);
    }
    if (params.active !== undefined) {
      httpParams = httpParams.set('active', params.active.toString());
    }

    return this.http.get<ApiResponse<Page<TaxInfo>>>(this.apiUrl, { params: httpParams });
  }

  getTaxInfo(id: number): Observable<ApiResponse<TaxInfo>> {
    return this.http.get<ApiResponse<TaxInfo>>(`${this.apiUrl}/${id}`);
  }

  createTaxInfo(taxInfo: TaxInfo): Observable<ApiResponse<TaxInfo>> {
    return this.http.post<ApiResponse<TaxInfo>>(this.apiUrl, taxInfo);
  }

  updateTaxInfo(id: number, taxInfo: TaxInfo): Observable<ApiResponse<TaxInfo>> {
    return this.http.put<ApiResponse<TaxInfo>>(`${this.apiUrl}/${id}`, taxInfo);
  }

  deleteTaxInfo(id: number): Observable<ApiResponse<void>> {
    return this.http.delete<ApiResponse<void>>(`${this.apiUrl}/${id}`);
  }

  toggleStatus(id: number): Observable<ApiResponse<TaxInfo>> {
    return this.http.put<ApiResponse<TaxInfo>>(`${this.apiUrl}/${id}/toggle-status`, {});
  }
}

File: src/main/java/com/fwai/turtle/service/interfaces/TaxInfoService.java
package com.fwai.turtle.service.interfaces;

import com.fwai.turtle.dto.TaxInfoDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface TaxInfoService {
    
    /**
     * 获取税务信息列表（分页）
     */
    Page<TaxInfoDTO> getTaxInfos(Pageable pageable, Boolean active);

    /**
     * 根据ID获取税务信息
     */
    TaxInfoDTO getTaxInfo(Long id);

    /**
     * 创建税务信息
     */
    TaxInfoDTO createTaxInfo(TaxInfoDTO taxInfoDTO);

    /**
     * 更新税务信息
     */
    TaxInfoDTO updateTaxInfo(Long id, TaxInfoDTO taxInfoDTO);

    /**
     * 删除税务信息
     */
    void deleteTaxInfo(Long id);

    /**
     * 切换税务信息状态
     */
    TaxInfoDTO toggleStatus(Long id);

    /**
     * 搜索税务信息
     */
    Page<TaxInfoDTO> searchTaxInfos(String query, Pageable pageable);
}
